src/main/java/com/example/SS2_Backend/model/onetomany/MatchingProblem.java

```diff
@@ -0,0 +1,362 @@
+package com.example.SS2_Backend.model.onetomany;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import lombok.extern.slf4j.Slf4j;
+import net.objecthunter.exp4j.Expression;
+import net.objecthunter.exp4j.ExpressionBuilder;
+import org.moeaframework.core.Problem;
+import org.moeaframework.core.Solution;
+import org.moeaframework.core.Variable;
+import org.moeaframework.core.variable.Permutation;
+
+import java.util.*;
+import java.util.function.DoubleUnaryOperator;
+import java.util.stream.DoubleStream;
+import java.util.stream.Stream;
+
+import static com.example.SS2_Backend.util.StringExpressionEvaluator.*;
+
+@Slf4j
+@Data
+@NoArgsConstructor
+@FieldDefaults(level = lombok.AccessLevel.PRIVATE)
+public class MatchingProblem implements Problem {
+
+    IndividualList individuals;
+    List<PreferenceList> preferenceLists;
+    int [][] excludedPairs;
+    PreferenceProvider preferencesProvider;
+    String evaluateFunctionForProviders;
+    String evaluateFunctionForConsumers;
+    String fitnessFunction;
+    boolean providerFunctionStatus = false;
+    boolean consumerFunctionStatus = false;
+    boolean fitnessFunctionStatus = false;
+
+    public void setPopulation(ArrayList<Individual> individuals, String[] propertiesNames, int [][] excludedPairs) {
+        this.individuals = new IndividualList(individuals, propertiesNames);
+        this.excludedPairs = excludedPairs;
+        initializeFields();
+    }
```

Comment:
should consider passing this in the *Constructor*

Reply:
Tạm thời merge luôn, sửa đổi vào pull request khác

Reply:
Nhắc Đạt chú ý đặt tên branch cho đúng rule